src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/tests/latest/test_bot_commands.py

```diff
@@ -293,6 +293,101 @@ def test_create_v4_js_webapp_bot(self, resource_group):
         # Delete bot
         self.cmd('az bot delete -g {rg} -n {botname}')
 
+    @ResourceGroupPreparer(random_name_length=20)
+    def test_botservice_create_should_remove_invalid_char_from_name_when_registration(self, resource_group):
+        bot_name = self.create_random_name(prefix='cli.', length=15)
+        self.kwargs.update({
+            'botname': bot_name,
+            'app_id': str(uuid.uuid4()),
+            'password': str(uuid.uuid4())
+        })
+
+        self.cmd('az bot create -k registration -g {rg} -n {botname} --appid {app_id} -p {password} '
+                 '-e https://testurl.com/api/messages',
+                 checks={
+                     self.check('resourceGroup', '{rg}'),
+                     self.check('type', 'Microsoft.BotService/botServices')
+                 })
+
+    @ResourceGroupPreparer(random_name_length=20)
+    def test_botservice_create_should_remove_invalid_char_from_name_when_webapp(self, resource_group):
+        bot_name = self.create_random_name(prefix='cli.', length=15)
```

Comment:
I believe this breaks CLI's recording process because the new name with `.` removed doesn't match any entry in the `create_random_name` table. 

Since I am bumping the subscription REST API version in https://github.com/Azure/azure-cli/pull/11886, I need to re-run all tests live but failed here. Did you record without `.` or manually edit the recording YAML: https://github.com/Azure/azure-cli/pull/8384/commits/03e5147c7e488ae36c01d3c92db827d8bfb1b9de#diff-4e739bb12b0f4e7932621ac65a0d6372R103

If so, we'd better change this test to unit test or live-only so that the test can follow the common process.

Reply:
As discussed with the author, the recording files were manually edited. This issue is addressed in #11935 by moving the tests into a `LiveScenarioTest`.